discovery_engine/pubspec.yaml

```
@@ -27,6 +27,9 @@ dependencies:
       url: https://github.com/xaynetwork/xayn_async_bindgen.git
       path: async_bindgen_dart_utils
       ref: a299c69aacb1f017c0f062a83203b5f6225c462f
+  source_gen: ^1.2.2
+  build: ^2.3.0
+  analyzer: ^3.4.1
```

❓ could we also use the latest `4.2.0` or is this conflicting with other deps or blue?
```suggestion
  analyzer: ^4.2.0
```

---

I'll give it a try. I think analyzer was already being used, and I just used the same version 🤔 but I may be mistaken

---

Yep seems like this doesn't work because of `freezed`:

```
Because freezed 1.1.1 depends on analyzer ^3.0.0 and no versions of freezed match >1.1.1
  <2.0.0, freezed ^1.1.1 requires analyzer ^3.0.0.
So, because xayn_discovery_engine depends on both analyzer ^4.2.0 and freezed ^1.1.1,
  version solving failed.
```

I wouldn't want to upgrade freezed as part of this PR as well, so it'll have to remain outdated for the time being 😅

---

iirc `freezed` is tied to that version because of blue, so we can't do much here. thx for giving it a try 👌